lib/amplitude_api.rb

```diff
@@ -198,5 +197,18 @@ def delete_body(user_ids, amplitude_ids, requester)
         }.delete_if { |_, value| value.nil? || value.empty? }
       )
     end
+
+    def segmentation_params(event, start_time, end_time, **options)
+      {
+        e: event.to_json,
+        m: options[:m],
+        start: start_time.strftime('%Y%m%d'),
+        end: end_time.strftime('%Y%m%d'),
+        i: options[:i],
+        s: (options[:s] || []).map(&:to_json),
+        g: options[:g],
+        limit: options[:limit]
+      }.compact
```

Comment:
this was `.delete_if { |_, value| value.nil? }` before but here is equivalent to [Hash#compact](https://ruby-doc.org/core-2.6.2/Hash.html#method-i-compact)

Reply:
This actually requires a deeper discussion of supported Ruby versions. The `compact` method for Hashes doesn't exist prior to version 2.4, so if anyone is still on version 2.3.x or prior, this function will always fail with a missing method error.

Version 2.4.0 was released about 2 years ago, so I would personally think it's safe to increase our minimum required Ruby version to that, but we'll need to change our test suite to remove older Ruby targets. We would also need to put at least a temporary note in the readme so that users updating to the latest version know we are dropping support for older Rubies.

Reply:
Thanks for the explanation!

Ruby 2.3 just got [EOL'd](https://www.ruby-lang.org/en/news/2019/03/31/support-of-ruby-2-3-has-ended/) i think we can consider dropping it.
If we decide to go for it I can add it to the changelog and the readme and fix travis, let me know.

Reply:
@toothrot I'm going to put in a PR to remove 2.3.x rubies from Travis since, as mentioned above, support officially ended March 31, 2019 (just a few weeks ago), so we might as well get it gone. Anyone that absolutely needs to rely on versions of Ruby prior to 2.4 can just pin an appropriate version of the gem.

Reply:
@toothrot @vbalazs I've added PR #48 to (a) remove any Ruby prior to 2.4 and (b) fix some rubocop stuff that started breaking after I did that.